### ISSUE TITLE
make CLIP-based dataloader  for imagenette

### ISSUE THREAD

**Issue by murphyk** [probml/pyprobml#465]
- Make a torch.utils.data.Dataset version of the [imagenette  dataset from TFDS](https://www.tensorflow.org/datasets/catalog/imagenette), which has 10 classes and ~10k train, ~4k val. (Use the 160 pixel version for speed.)
- Take a [pre-trained CLIP pytorch model from openai](https://github.com/openai/CLIP), and use it as a feature extractor (`features = model.encode_image(image_inputs)`) to make a torchvisions.transforms class. (The "ViT-B/32" model returns features of size 512 for each image. The "RN50" mode uses 1024 dimensions.)
-  Compose the above to  make a dataloader for imagenette that returns low-dim version of the images. (We will use this in future demos.)
 - Repeat the above for the [stanford online products dataset from TFDS](https://www.tensorflow.org/datasets/catalog/stanford_online_products), which is much larger (~60k train, ~60k test, 22634 classes, image size is unknown, but file size is a reasonable 3GB).
 - Repeat the above for the [coco detection dataset in torchvision](https://pytorch.org/vision/stable/datasets.html#detection). This is ~200k images (size unknown), with many kinds of annotation.

**Comment by murphyk** [comment]
moved to hermes
